libraries/app/database_api.cpp

```diff
@@ -1888,6 +1893,28 @@ std::string database_api::get_transaction_hex_without_sig(
 {
    return my->get_transaction_hex_without_sig(trx);
 }
+optional<query_trx_info> database_api_impl::get_transaction_by_txid(transaction_id_type txid)const
+{
+#ifdef QUERY_TXID_PLUGIN_ABLE
```

Comment:
If the plugin is not enabled the method doesn't return anything. IMO it should throw.